File: core/src/main/java/com/caibowen/prma/core/filter/basic/LevelFilter.java
package com.caibowen.prma.core.filter.basic;

import com.caibowen.prma.api.LogLevel;

/**
 * @author BowenCai
 * @since 11-11-2014.
 */
public class LevelFilter extends AbstractFilter<LogLevel> {


    @Override
    protected int doAccept(LogLevel logLevel) {
        return 0;
    }


}

File: core/src/main/scala/com/caibowen/prma/core/filter/LevelFilter.scala
package com.caibowen.prma.core.filter

import com.caibowen.prma.api.LogLevel

/**
 * @author BowenCai
 * @since  01/12/2014.
 */
class LevelFilter(lower: Int, upper: Int) extends Filter[LogLevel] {

  def this(ls: String, us: String) {
    this(LogLevel.from(ls).levelInt, LogLevel.from(us).levelInt)
  }

  override def doAccept(le: LogLevel): Int = {
    if (lower <= le.levelInt && le.levelInt <= upper)
      1
    else
      -1
  }

}
